test/test_autograd.py

```diff
@@ -3903,6 +3903,28 @@ def test_leaky_relu_inplace_with_neg_slope(self):
             with self.assertRaisesRegex(RuntimeError, "call out-of-place version"):
                 b.backward(torch.ones(2, device=device))
 
+    def test_custom_function_local_inplace(self):
+        class MyFn(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, inp, inplace):
+                view = inp.clone()[:3]
+                if inplace:
+                    view += 2
+                return view
+
+            @staticmethod
+            def backward(ctx, grad):
+                return grad, None
+
+        base = torch.rand(10, requires_grad=True)
+
+        foo = MyFn.apply(base, False)
+        self.assertEqual(foo.grad_fn.__class__.__name__, "MyFnBackward")
+
+        foo = MyFn.apply(base, True)
+        self.assertEqual(foo.grad_fn.__class__.__name__, "MyFnBackward")
```

Comment:
is the problem printing the grad function or that some behavior was wrong?

Reply:
The behavior was wrong: even though neither the view or the base was modified after the graph was created, we still rewrote the graph as if the inplace happened after the graph creation.

This is not ok in particular because rewriting this history for things that are views of temporary Variables is dangerous (plain wrong but for now we just warn).